File: core/MinimumDifferenceBetweenHighestAndLowestOfKScores/MinimumDifferenceBetweenHighestAndLowestOfKScores.java
package com.dbc;

import java.util.Arrays;

public class MinimumDifferenceBetweenHighestAndLowestOfKScores {
    public int minimumDifference(int[] nums, int k) {
        Arrays.sort(nums);
        int point = k - 1, res = 100000;
        while (point < nums.length) {
            res = Math.min(res, nums[point] - nums[point - k + 1]);
            point++;
        }
        return res;
    }
}

File: core/MinimumDifferenceBetweenHighestAndLowestOfKScores/MinimumDifferenceBetweenHighestAndLowestOfKScores.py
from typing import List
import heapq


def minimumDifference(nums: List[int], k: int) -> int:
    nums.sort()
    right, res = k - 1, 100000
    while right < len(nums):
        res = min(res, nums[right] - nums[right - k + 1])
        right += 1

    return res


if __name__ == '__main__':
    print(minimumDifference([9, 4, 1, 7], 2))
